### ISSUE TITLE
[Feature]: How to set the checkbox to display continuously, even if it exceeds the canvas, it can still be displayed normally

### ISSUE THREAD

**Issue by zhenzi0322** [fabricjs/fabric.js#10166]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

![image](https://github.com/user-attachments/assets/992cfc37-cb5c-4170-9394-068ba960ca84)

### Current State

What I want is for the selection box beyond the canvas to be displayed normally

**Comment by houssameddine-h** [comment]
I don't think this would be possible, because it's obviously outside of the canvas, and we can't draw shapes outside of it.

I did achieve something similar by making the actual canvas big, and drawing a locked rectangle in it. When exporting the canvas, I only export the content within the locked rectangle.

![Screenshot 2024-09-27 at 14-46-15 React App](https://github.com/user-attachments/assets/01320cd8-3adb-4e54-8d0d-141f06933049)

**Comment by zhenzi0322** [comment]
What I want is a rectangle that can be moved out of the canvas to see the control box

**Comment by houssameddine-h** [comment]
So I kind of forgot about this feature, and then today, while playing with some design tools, I thought I might need this at some point. So this is how I did it:

First, I created 4 locked rectangles around the main drawing area, and made them extend to the edges of the canvas:

![Screenshot 2024-10-04 at 15-21-53 React App](https://github.com/user-attachments/assets/f20ea6f9-7c5e-4ac3-9bf2-63834cbee928)

Then, I set the fill color of these rectangles to my desired background color of the canvas:

![Screenshot 2024-10-04 at 15-26-35 React App](https://github.com/user-attachments/assets/8c3332cd-950a-4efb-85d8-37c0adaf3bba)

Finally, I moved the rectangles to the front of the canvas stack (made them the last layer).

[screen-recorder-video-oct-04-2024-15-34-32.webm](https://github.com/user-attachments/assets/acf9e3bf-053f-4601-9234-ec5a8b5f2885)

The tricky part is whenever you add an object to the canvas, you have to move the rectangles to the front of the stack.

I know this is not exactly what you want, but it might be useful for someone else.
